asyncscope.md

```diff
@@ -819,6 +819,166 @@ _See_ [Why does `async_scope` terminate in the destructor instead of blocking li
 > NOTE: async RAII could be used to signal early completion because it would be composed with other async operation lifetimes.
 > The operation being stopped would complete before the async RAII operation completed -- without any blocking.
 
+Alternative design
+==================
+
+This section describes an alternative design that would break down `async_scope` into smaller parts, providing better composability and finer-grained control.
+
+## Definitions
+```cpp
+struct async_scope {
+    async_scope();
+    ~async_scope();
+    async_scope(const async_scope&) = delete;
+    async_scope(async_scope&&) = delete;
+    async_scope& operator=(const async_scope&) = delete;
+    async_scope& operator=(async_scope&&) = delete;
+
+    template <sender S>
+    @_nest-sender_@<S> nest(S&& snd);
+
+    template <sender S>
+    [[nodiscard]]
+    @_on-empty-sender_@<S> when_empty(S&& snd) const noexcept;
+};
+```
+
+The meaning of `nest()` and `when_empty()` is kept the same.
+All other methods that are absent in this alternative design can be built on top of the remaining functionality.
+
+### `spawn()`
+
+If `scope` is an object of type `async_scope`, then `scope.spawn(snd)` is equivalent with:
+```c++
+start_detached(scope.nest(snd));
+```
+
+### `spawn_future()`
+
+If `scope` is an object of type `async_scope`, then `scope.spawn_future(snd)` is equivalent with:
+```c++
+ensure_started(scope.nest(snd));
+```
+
+### `on_empty()`
+
+If `scope` is an object of type `async_scope`, then `scope.on_empty()` is equivalent with:
+```c++
+scope.when_empty(just());
+```
+
+### Cancellation
+
+In this alternative, `async_scope` doesn't provide direct support for cancellation.
+However, this can be added on top.
+If `scope` is an object of type `async_scope`, and `stoken` is a stop token, then we can build cancellation in with the following:
+```c++
+scope.nest(stop_when(snd, stoken))
```

Comment:
I think we would want to see the cancellation path here too.

If we are going to wait for the scope to complete we need to decide what happens - do we stop accepting work (which cancellation tied to the scope can allow) and cancel in-flight work so we can guarantee safe shutdown? 

Kirk's recent work on a phased async_scope makes this a bit more explicit.

Reply:
In this alternative, you loose the ability co cancel any future work (unless everything is spawn using the construct above). This may be an inconvenience, but not all flows might have cases in which work is spawned when we are trying to end the scope. It provides a bit more granular access to the user to decide whether global cancellation is needed, or not.

If global cancellation is needed, then the user can always make a small abstraction on top of `async_scope` that always packs the stop token with the work needing to be spawned.

I guess, in an ideal world, we would propose two abstractions:
* a low level abstraction, similar to the one in this alternative
* a friendlier version of `async_scope`, as described by the main paper